napari/_qt/qt_dims_slider.py

```diff
@@ -0,0 +1,471 @@
+from typing import Optional, Tuple
+
+import numpy as np
+from qtpy.QtCore import Qt, QTimer, Signal, Slot, QObject
+from qtpy.QtWidgets import QApplication
+from qtpy.QtWidgets import (
+    QComboBox,
+    QHBoxLayout,
+    QLabel,
+    QLineEdit,
+    QPushButton,
+    QDoubleSpinBox,
+    QWidget,
+    QCheckBox,
+)
+
+from ..util.event import Event
+from .qt_modal import QtModalPopup
+from .qt_scrollbar import ModifiedScrollBar
+from .util import new_worker_qthread
+
+
+class QtDimSliderWidget(QWidget):
+    """Compound widget to hold the label, slider and play button for an axis.
+
+    These will usually be instantiated in the QtDims._create_sliders method.
+    This widget *must* be instantiated with a parent QtDims.
+    """
+
+    label_changed = Signal(int, str)  # axis, label
```

Comment:
so I guess this relates to #730, and if not necessary would simplify the code right? We should figure this out from @royerloic because it is quite a bit more overhead if not needed right @tlambert03 and now would probably be a good time to make that change? or could it wait?

Reply:
It’s as reasonable a time as any to trim it, but it could definitely wait.  I don’t know whether it costs us much (other than making it slightly less clear which signals actually communicates with other processes).  My very slight preference would be to omit those signals until used (premature optimization?), but would absolutely defer to @royerloic on that

Reply:
Based on the comment from @royerloic in #730 it seems like this can be simplified here. I’d probably do this before we merge

Reply:
I’d also simplify the dims view too

Reply:
I saw you were able to drop the signals from `QtDims`, which I think simplified it quite a bit. Did you need to keep these ones because they also are used to communicate with `QtDims` whereas the other ones were entirely internal to that class?

Reply:
Yeah, those remaining signals are used by other `QObjects` to communicate with `QtDims`, that's the distinction that I generally think determines whether we want to use a signal or call an internal method.

Reply:
makes sense, thanks for helping me understand!